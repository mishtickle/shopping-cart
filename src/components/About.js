import MadeNZ from '../img/madeInNewZealand.jpg'

function About() {
  return (
    <>
        <nav>
        <Link to="/">Home</Link>
        <Link to="/shop">Shop</Link>
        <Link to="/contact">Contact</Link>
        </nav>
        <br></br>
        <br></br>
        <br></br>
        <div className="aboutContainer">
            <h1>About Us:</h1>
            <h3>We search far and wide for the best solar powered gadgets</h3>
            <h3>Proudly New Zealand owned and operated</h3>
            <img src={MadeNZ} alt="Made in New Zealand" height="300px"></img>
        </div>
    </>
  );
}

export default About;
