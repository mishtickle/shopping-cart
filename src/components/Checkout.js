import { StateValues } from "../RouteSwitch.js";
import React, { useContext, useEffect } from 'react';
import { Link } from "react-router-dom"

function Checkout(props) {

  const { 
    light, setLight, 
    radio, setRadio, 
    lamp, setLamp, 
    shower, setShower, 
    value, setValue } = React.useContext(StateValues);

  useEffect(() => {
    setValue(light*30 + radio*50 + lamp*20 + shower*20)
  }, [light, radio, lamp, shower, value])
  
  return (
      <>
          <nav>
              <Link to="/">Home</Link>
              <Link to="/shop">Shop</Link>
              <Link to="/about">About</Link>
              <Link to="/contact">Contact</Link>
          </nav>
          <br></br>
          <br></br>
          <br></br>
          <h3>Checkout:</h3>
          {light > 0 ? <p>Light <br></br>Quantity: {light} <br></br> Price: ${light * 30} <br></br> <button onClick={() => setLight(0)}>Remove</button> </p> : ''}
          {radio > 0 ? <p>Radio <br></br>Quantity: {radio} <br></br> Price: ${radio * 50} <br></br> <button onClick={() => setRadio(0)}>Remove</button></p> : ''}
          {lamp > 0 ? <p>Lamp <br></br>Quantity: {lamp} <br></br> Price: ${lamp * 20} <br></br> <button onClick={() => setLamp(0)}>Remove</button></p> : ''}
          {shower > 0 ? <p>Shower <br></br>Quantity: {shower} <br></br> Price: ${shower * 20} <br></br> <button onClick={() => setShower(0)}>Remove</button></p> : ''}
          {value == 0 ? <p>It looks like you don't have anything in your cart.  <br></br>Visit the shop for some cool products.</p> : ''}
          <br></br>
          <br></br>
          {value > 0? <><h3>Total Value: ${value}</h3></> : ''}
          {value > 0 ? <button>Pay Now</button> : ""}

          

      </>
    );
  }
  
  export default Checkout;
  