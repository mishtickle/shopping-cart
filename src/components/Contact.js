import { Link } from 'react-router-dom'

function Contact() {
  return (
    <>
        <nav>
            <Link to="/">Home</Link>
            <Link to="/shop">Shop</Link>
            <Link to="/about">About</Link>
        </nav>
        <br></br>
        <br></br>
        <br></br>
        <div className="contactContainer">
            <h1>Contact:</h1>
            <h3>notarealemail@email.com</h3>
        </div>
    </>
  );
}

export default Contact;
