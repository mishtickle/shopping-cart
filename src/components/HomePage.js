import Earth from '../img/earth.jpg'
import { Link } from "react-router-dom"
import { MemoryRouter } from 'react-router-dom';

function HomePage() {
  return (
    <>
        <nav>
          <Link to="/shop">Shop</Link>
          <Link to="/contact">Contact</Link>
          <Link to="/about">About</Link>
        </nav>
        <br></br>
        <br></br>
        <br></br>
        <div className="background-image">
            <img src={Earth} alt="Earth" width='1263px'></img>
            <h1 className="solarHeading">The Solar Shop</h1>
        </div>
    </>
  );
}

export default HomePage;
