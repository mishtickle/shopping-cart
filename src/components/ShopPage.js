import './shopPage.css';
import solarRadio from '../img/solarRadio.jpg'
import solarFlashLight from '../img/solarFlashLight.jpg'
import solarReadingLamp from '../img/solarReadingLamp.jpg'
import solarShower from '../img/solarShower.jpg'
import React, { useEffect } from 'react';
import { StateValues } from '../RouteSwitch.js';
import { Link } from 'react-router-dom'

function ShopPage(props) {
    const { light, radio, shower, lamp, items, value, incrementRadio, decrementRadio, changeRadio,
    incrementLight, decrementLight, changeLight,
    incrementShower, decrementShower, changeShower,
    incrementLamp, decrementLamp, changeLamp,
    setItems, setValue } = React.useContext(StateValues);

    useEffect(() => {
        setItems(Number(radio) + Number(light) + Number(lamp) + Number(shower));
        setValue(radio*50 + light*30 + lamp*20 + shower*20)
      }, [radio, light, lamp, shower])

  return (
    <>
        <nav>
            <Link to="/">Home</Link>
            <Link to="/contact"> Contact</Link>
            <Link href="/about"> About</Link>
        </nav>
        <br></br>
        <br></br>
        <h1 className="shopHeading">Shop</h1>
        <div className="container">
            <div className="products">
                <div className="product1">
                    <img src={solarRadio} alt='Solar Radio' width='200px' height='200px'></img>
                    <h3>Solar Powered Radio</h3>
                    <h3>$50.00</h3>
                    <input type="text" className="radio" value={radio} onChange={changeRadio}></input>
                    <button onClick={incrementRadio}>+</button>
                    <button onClick={decrementRadio}>-</button>
                    <br></br>
                    <br></br>
                </div>
                <div className="product2">
                    <img src={solarFlashLight} alt='Solar Flash Light' width='200px' height='200px'></img>
                    <h3>Solar Powered Flash Light</h3>
                    <h3>$30.00</h3>
                    <input type="text" className="flashlight" value={light} onChange={changeLight}></input>
                    <button onClick={incrementLight}>+</button>
                    <button onClick={decrementLight}>-</button>
                    <br></br>
                    <br></br>
                </div>
                <div className="product3">
                    <img src={solarShower} alt='Solar Shower' width='200px' height='200px'></img>
                    <h3>Solar Powered Shower</h3>
                    <h3>$20.00</h3>
                    <input type="text" className="shower" value={shower} onChange={changeShower}></input>
                    <button onClick={incrementShower}>+</button>
                    <button onClick={decrementShower}>-</button>
                    <br></br>
                </div>
                <div className="product4">
                    <img src={solarReadingLamp} alt='Solar Reading Lamp' width='200px' height='200px'></img>
                    <h3>Solar Powered Reading Lamp</h3>
                    <h3>$20.00</h3>
                    <input type="text" className="lamp" value={lamp} onChange={changeLamp}></input>
                    <button onClick={incrementLamp}>+</button>
                    <button onClick={decrementLamp}>-</button>
                    <br></br>
                </div>
            </div>
            <div className="cart">
                { items > 0 ? <>
                    <h4>Cart</h4>
                    <h5>Items: {items}</h5>
                    <h5>Value: ${value}</h5>
                    <button><Link to="/checkout" className="checkout">Purchase</Link></button>
                </> : ''}
                
                
            </div>
        </div>
    </>
  );
}

export default (ShopPage);
