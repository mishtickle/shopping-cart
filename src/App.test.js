import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import HomePage from "./components/HomePage.js";
import About from "./components/About.js";
import Checkout from "./components/Checkout.js";
import Contact from "./components/Contact.js";
import ShopPage from "./components/ShopPage.js";

describe("Home page component", () => {
  it("renders the home page", () => {
    const { container } = render(<HomePage />, {wrapper: BrowserRouter}) ;
    expect(container).toMatchSnapshot();
  })
})

describe("Contact page component", () => {
  it("renders the contact page", () => {
    const { container } = render(<Contact />, {wrapper: BrowserRouter}) ;
    expect(container).toMatchSnapshot();
  })
})

describe("About page component", () => {
  it("renders the about page", () => {
    const { container } = render(<About />, {wrapper: BrowserRouter}) ;
    expect(container).toMatchSnapshot();
  })
})