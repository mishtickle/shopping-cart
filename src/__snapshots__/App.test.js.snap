// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page component renders the about page 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      Home
    </a>
    <a
      href="/shop"
    >
      Shop
    </a>
    <a
      href="/contact"
    >
      Contact
    </a>
  </nav>
  <br />
  <br />
  <br />
  <div
    class="aboutContainer"
  >
    <h1>
      About Us:
    </h1>
    <h3>
      We search far and wide for the best solar powered gadgets
    </h3>
    <h3>
      Proudly New Zealand owned and operated
    </h3>
    <img
      alt="Made in New Zealand"
      height="300px"
      src="madeInNewZealand.jpg"
    />
  </div>
</div>
`;

exports[`Contact page component renders the contact page 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      Home
    </a>
    <a
      href="/shop"
    >
      Shop
    </a>
    <a
      href="/about"
    >
      About
    </a>
  </nav>
  <br />
  <br />
  <br />
  <div
    class="contactContainer"
  >
    <h1>
      Contact:
    </h1>
    <h3>
      notarealemail@email.com
    </h3>
  </div>
</div>
`;

exports[`Home page component renders the home page 1`] = `
<div>
  <nav>
    <a
      href="/shop"
    >
      Shop
    </a>
    <a
      href="/contact"
    >
      Contact
    </a>
    <a
      href="/about"
    >
      About
    </a>
  </nav>
  <br />
  <br />
  <br />
  <div
    class="background-image"
  >
    <img
      alt="Earth"
      src="earth.jpg"
      width="1263px"
    />
    <h1
      class="solarHeading"
    >
      The Solar Shop
    </h1>
  </div>
</div>
`;
